### ISSUE TITLE
SSH ProxyJump after commit bbfc31fc is broken

### ISSUE THREAD

**Issue by artizirk** [Fizzadar/pyinfra#714]
## Describe the bug
`ProxyJump` from `~/.ssh/config` is correctly picked up but it fails when connecting to the target.

Looks like [SSHClient.gateway](https://github.com/Fizzadar/pyinfra/blob/af4bc73e74b367fe53180b1899af6b8b6d3e5561/pyinfra/api/connectors/sshuserclient/client.py#L49) wants to access `self.hostname` and `self.config` that were removed in commit bbfc31fc

## To Reproduce
1.
```
# ~/.ssh/config
Host mammut
  ProxyJump kodu.arti.ee
```
2.
```
pyinfra mammut exec -- hostname
```
3.

```
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/host.py", line 159, in connect
    self.connection = self.executor.connect(self.state, self)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/ssh.py", line 220, in connect
    client.connect(hostname, **kwargs)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 37, in connect
    hostname, config, forward_agent = self.parse_config(hostname)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 95, in parse_config
    sock = c.gateway(target, target_config['port'])
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 54, in gateway
    (self.hostname, self.config['port']))
AttributeError: 'SSHClient' object has no attribute 'hostname'
2021-11-30T16:09:17Z <Greenlet at 0x7f4813b54150: <bound method Host.connect of Host(mammut)>> failed with AttributeError

--> An unexpected internal exception occurred:

  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 54, in gateway
    (self.hostname, self.config['port']))
AttributeError: 'SSHClient' object has no attribute 'hostname'

--> The full traceback has been written to pyinfra-debug.log
--> If this is unexpected please consider submitting a bug report on GitHub, for more information run `pyinfra --support`.
```

## Expected behavior
No errors

## Meta
+ Include output of `pyinfra --support`.
```
--> Support information:

    If you are having issues with pyinfra or wish to make feature requests, please
    check out the GitHub issues at https://github.com/Fizzadar/pyinfra/issues .
    When adding an issue, be sure to include the following:

    System: Linux
      Platform: Linux-5.15.5-arch1-1-x86_64-with-glibc2.33
      Release: 5.15.5-arch1-1
      Machine: x86_64
    pyinfra: v1.5
    Executable: /home/arti/tmp/scratch-1635932696/venv/bin/pyinfra
    Python: 3.9.7 (CPython, GCC 11.1.0)
```

+ How was pyinfra installed (source/pip)?

pip

+ Include pyinfra-debug.log (if one was created)

```
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra_cli/main.py", line 236, in cli
    _main(*args, **kwargs)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra_cli/main.py", line 548, in _main
    connect_all(state)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connect.py", line 35, in connect_all
    greenlet.get()
  File "src/gevent/greenlet.py", line 803, in gevent._gevent_cgreenlet.Greenlet.get
  File "src/gevent/greenlet.py", line 371, in gevent._gevent_cgreenlet.Greenlet._raise_exception
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/gevent/_compat.py", line 65, in reraise
    raise value.with_traceback(tb)
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/host.py", line 159, in connect
    self.connection = self.executor.connect(self.state, self)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/ssh.py", line 220, in connect
    client.connect(hostname, **kwargs)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 37, in connect
    hostname, config, forward_agent = self.parse_config(hostname)
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 95, in parse_config
    sock = c.gateway(target, target_config['port'])
  File "/home/arti/tmp/scratch-1635932696/venv/lib/python3.9/site-packages/pyinfra/api/connectors/sshuserclient/client.py", line 54, in gateway
    (self.hostname, self.config['port']))
AttributeError: 'SSHClient' object has no attribute 'hostname'
```

**Comment by artizirk** [comment]
seems to be fixed in 5c14d4391bafb17daba4d3eb6e4e0bd24c7741bf

**Comment by Fizzadar** [comment]
@artizirk thank you for reporting this - indeed it should now be fixed and I've just released it in `v1.6[.1]`.
